corehq/change_ownership.py

```diff
@@ -0,0 +1,210 @@
+import math
+import os
+from datetime import datetime
+
+from casexml.apps.case.mock import CaseBlock
+from dimagi.utils.chunked import chunked
+
+from corehq.apps.es import CaseSearchES
+from corehq.apps.es.cases import case_type
+from corehq.apps.hqcase.utils import submit_case_blocks
+from corehq.apps.locations.models import SQLLocation
+from corehq.apps.users.models import CommCareUser
+from corehq.form_processor.models import CommCareCase
+from corehq.util.log import with_progress_bar
+
+current_time = datetime.now().time()
+# TODO: Need to incorporate a format for success logs to make it easy to reverse
+success_case_log_file_path = os.path.expanduser(f'~/script_success_{current_time}.log')  # Use this if we ever need to rollback changes
+skipped_log_file_path = os.path.expanduser(f'~/script_skipped_{current_time}.log')
+error_log_file_path = os.path.expanduser(f'~/script_error_{current_time}.log')
+
+class Updater(object):
+    batch_size = 100  # Could potentially increase this
+    domain = 'alafiacomm'
+    stat_counts = {
+        'success': 0,
+        'skipped': 0,
+        'failed': 0,
+    }
+
+    def write_to_log(self, file_path, ids, message=''):
+        # TODO: Add ability to write a reverse ID for reversing action
+        msg_out = f'|Reason: {message}' if message else ''
+        with open(file_path, 'a') as log:
+            for id in ids:                
+                log.write(f'{id} {msg_out}\n')
+            log.close()
```

Comment:
this is going to be lots of file operations. Can we consider keeping the file open with a `with` block and if needed use `flush` to write to file though the writer might do it automatically for you. `csv.writer` might do it automatically for you.

Reply:
Noting that this is no longer relevant since the script has moved away from logging into a file to storing all the necessary data in a SQLite DB.